ui/src/components/Pin.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import L from "leaflet";
+import { Marker } from "react-leaflet";
+import resting from "../assets/resting.svg";
+import unfocused from "../assets/unfocused.svg";
+
+export enum PinState {
+  Resting,
+  Selected,
+  Unfocused,
+}
+
+export interface PinProps {
+  state?: PinState;
+  position: [number, number];
+}
+
+export function Pin({ state = PinState.Resting, position }: PinProps) {
```

Comment:
Can you add `JSX.Element` as a return type here?

You can search the codebase for `JSX.Element` to see how it is used.